src/frontend/src/components/forms/ApiForm.tsx

```diff
@@ -163,13 +172,23 @@ export function OptionsApiForm({
  * based on an API endpoint.
  */
 export function ApiForm({ id, props }: { id: string; props: ApiFormProps }) {
-  const defaultValues: FieldValues = useMemo(
-    () =>
-      mapFields(props.fields ?? {}, (_path, field) => {
-        return field.default ?? undefined;
-      }),
-    [props.fields]
-  );
+  const defaultValues: FieldValues = useMemo(() => {
+    let defaultValuesMap = mapFields(props.fields ?? {}, (_path, field) => {
+      return field.value ?? field.default ?? undefined;
+    });
+
+    // If the user has specified initial data, use that instead
+    if (props.initialData) {
+      defaultValuesMap = {
+        ...defaultValuesMap,
+        ...props.initialData
+      };
+    }
```

Comment:
Does this work recursively for nested forms?

Reply:
Probably not - at least I have not tested it with that. Do you have any examples of a nested form already implemented I could test it with?

Reply:
Yes, the part form:

https://github.com/inventree/InvenTree/blob/master/src/frontend/src/forms/PartForms.tsx#L72-L84

Reply:
Ok, yep works just fine for nested. I'll add to the playground example.

Reply:
But not if you only supply just one value and get the other from the fields default, because the spread syntax doesn't do deepmerging. But given the fact that one probability doesn't use this like this it's fine for me.